File: Source/BandClickBackend/BandClickBackend.Domain/Entities/MetronomeSettings.cs
using System;
using System.Collections.Generic;
using BandClickBackend.Domain.Common;

namespace BandClickBackend.Domain.Entities
{
    public class MetronomeSettings : ShareableEntity<MetronomeSettingsRaiting>
    {
        public string Name { get; set; }
        public int NumberOfMeasures { get; set; }
        public int Tempo { get; set; }
        public Metre Metre { get; set; }
        public MetronomeSettingsType Type { get; set; }
        public ICollection<MetronomeSettingsComment> Comments { get; set; }
        public ICollection<MetronomeSettingsInPlaylist> Playlists { get; set; }
    }
}

File: Source/BandClickFrontend/band-click-frontend/src/models/MetronomeSettings/MetronomeSettings.ts
import { CommentInfo } from "../Comments/Comment";
import { Metre } from "../Metre/Metre";
import { MetronomeSettingsType } from "./MetronomeSettingsType";

export interface MetronomeSettings {
    id: string,
    name: string,
    numberOfMeasures: number,
    tempo: number,
    metre: Metre,
    type: MetronomeSettingsType,
    comments: CommentInfo[],
    isShared: boolean,
    positiveRaitingCount: number,
    negativeRaitingCount: number,
    author: string,
    commentsCount: number,
    created: Date
}
